subxt/src/rpc.rs

```diff
@@ -355,6 +355,23 @@ impl<T: Config> Rpc<T> {
         }
     }
 
+    /// Get a block hash given a T::BlockNumber from our config.
+    /// This may not be useful externally, but we need to be able to
+    /// get detail for T::BlockNumbers.
+    #[doc(hidden)]
+    pub async fn block_hash_internal(
+        &self,
+        block_number: T::BlockNumber,
+    ) -> Result<Option<T::Hash>, BasicError> {
+        let block_number = ListOrValue::Value(block_number);
+        let params = rpc_params![block_number];
+        let list_or_value = self.client.request("chain_getBlockHash", params).await?;
+        match list_or_value {
+            ListOrValue::Value(hash) => Ok(hash),
+            ListOrValue::List(_) => Err("Expected a Value, got a List".into()),
+        }
+    }
```

Comment:
I'm not a fan of duplicating this `block_hash` function. The problem is that `Config::BlockNumber` isn't something that we can easily work with.

I could also require that `Config::BlockNumber`implements something like `Into<u128>`, or I could remove the "other" copy of this function, but it sounds like some consideration went into it, so I was hesitant. What do you think @ascjones?

Reply:
(the nice thing abour requiring `Into<u128>` or similar would be that I can get rid of `num_traits` again, and the other required traits I added to `BlockNumber`, which I was just using to add a minimal amount of information about the BlockNumber so that I could iterate over them.)

Reply:
I think it is reasonable to use `Into<u128>`

Reply:
I went for `Into<u64>` in the end because u128's don't serialize to JSON so easily; I'm hoping `u64` is also a safe target here (I assume it's basically a choice between `u32` and `u64` that people would tend to make?)

Reply:
`Into<u64>` should also be fine.